From 1e9863e2ae8466ba3abe9f44787d1ac5770edf19 Mon Sep 17 00:00:00 2001
From: Andrew de los Reyes <adlr@chromium.org>
Date: Mon, 20 Feb 2012 17:22:13 -0800
Subject: [PATCH 1/3] Take action on KeyPress, not KeyRelease

When using KeyRelease, it feels like there is a delay between using
the keyboard and cursor disappearing, which makes it feel unpolished.
---
 unclutter.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/unclutter.c b/unclutter.c
index 23bb99b..ab4e039 100644
--- a/unclutter.c
+++ b/unclutter.c
@@ -180,7 +180,7 @@ main(argc,argv)char **argv;{
 	    realroot[screen] = XRootWindow(display,screen);
 	    cursor[screen] = createnullcursor(display,realroot[screen]);
 	    if(idletime<0)
-		XSelectInput(display,realroot[screen],KeyReleaseMask);
+		XSelectInput(display,realroot[screen],KeyPressMask);
 	}
     screen = DefaultScreen(display);
     root = VirtualRootWindow(display,screen);
@@ -209,7 +209,7 @@ main(argc,argv)char **argv;{
 		XEvent event;
 		do{
 		    XNextEvent(display,&event);
-		}while(event.type != KeyRelease ||
+		}while(event.type != KeyPress ||
 		       (event.xkey.state & ANYBUTTON));
 		oldx = event.xkey.x_root;
 		oldy = event.xkey.y_root;
@@ -265,7 +265,7 @@ main(argc,argv)char **argv;{
 		XEvent event;
 		do{
 		    XNextEvent(display,&event);
-		}while(event.type==KeyRelease ||
+		}while(event.type==KeyPress ||
 		       (event.type==MotionNotify &&
 			ALMOSTEQUAL(rootx,event.xmotion.x) &&
 			ALMOSTEQUAL(rooty,event.xmotion.y)));
-- 
1.7.7.3

